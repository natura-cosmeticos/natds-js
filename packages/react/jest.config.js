const base = require('../../jest.config.base.js')
const { displayName, name } = require('./package.json')

module.exports = {
  ...base,
  collectCoverageFrom: [
    'src/**/*.{js,jsx,ts,tsx}',
    '!src/**/*.stories.tsx',
    '!src/**/*.subcomponent.tsx',
    '!src/**/*.styles.ts',
    '!src/Templates/**',
    '!src/growthPlanColors/**',
    '!src/helpers/**'
  ],
  errorOnDeprecated: true,
  displayName,
  name,
  testEnvironment: 'jsdom',
  setupFilesAfterEnv: ['<rootDir>/setupTests.js'],
  roots: ['<rootDir>/src'],
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
    '^.+\\.mdx?$': '@storybook/addon-docs/jest-transform-mdx',
    '\\.(jpg|jpeg|png|gif|webp|svg)$': 'jest-transform-file'
  }
}
