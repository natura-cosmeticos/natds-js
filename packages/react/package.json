{
  "name": "@naturacosmeticos/natds-react",
  "displayName": "NatDS-React",
  "description": "A collection of components from Natura Design System for React",
  "version": "2.66.0-alpha.DSY-5577.5.0",
  "private": false,
  "keywords": [
    "design-system",
    "natura",
    "react",
    "web"
  ],
  "homepage": "https://github.com/natura-cosmeticos/natds-js/tree/master/packages/react/README.md",
  "bugs": {
    "url": "https://github.com/natura-cosmeticos/natds-js/issues"
  },
  "license": "ISC",
  "author": {
    "email": "designsystem@natura.net",
    "name": "Natura&Co.",
    "url": "https://github.com/natura-cosmeticos"
  },
  "files": [
    "README.md",
    "dist/"
  ],
  "jsdelivr": "dist/umd/index.js",
  "source": "src/index.ts",
  "types": "dist/index.d.ts",
  "typings": "dist/index.d.ts",
  "umd:main": "dist/umd/index.js",
  "unpkg": "dist/umd/index.js",
  "main": "dist/index.js",
  "directories": {
    "lib": "dist"
  },
  "repository": {
    "directory": "packages/react",
    "type": "git",
    "url": "git+https://github.com/natura-cosmeticos/natds-js.git"
  },
  "scripts": {
    "build": "yarn clean && tsc -p ./tsconfig.json && yarn rollup:build",
    "build:storybook": "yarn build-storybook -o docs/dist",
    "build:storybook:ci": "yarn build:storybook --quiet",
    "clean": "rm -rf ./dist",
    "copy:assets": "mkdir -p ./dist/assets && cp -r ./src/assets/* ./dist/assets",
    "lint": "eslint --ext .js,.jsx,.ts,.tsx ./src --color",
    "lint:fix": "yarn lint --fix",
    "rollup:build": "rollup -c",
    "storybook": "start-storybook --no-manager-cache -p 6006",
    "test": "jest --testPathPattern=src --verbose",
    "test:ci": "jest  --testPathPattern=src --coverage && codecov -t",
    "test:regression": "loki",
    "test:watch": "jest --watch --detectOpenHandles",
    "test:clean": "jest --update-snapshot",
    "generate": "yarn plop --plopfile ./generators/plopfile.js"
  },
  "dependencies": {
    "@naturacosmeticos/natds-icons": "1.28.1",
    "@naturacosmeticos/natds-themes": "0.83.3",
    "react-jss": "10.9.0"
  },
  "devDependencies": {
    "@types/react": "17.0.39",
    "@types/react-dom": "17.0.13",
    "plop": "3.0.5",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-test-renderer": "17.0.2",
    "storybook-dark-mode": "1.0.9"
  },
  "peerDependencies": {
    "@types/react": ">= 16.8.6",
    "@types/react-dom": "17.0.13",
    "react": ">= 16.8.4",
    "react-dom": ">= 16.8.4"
  },
  "peerDependenciesMeta": {
    "@types/react": {
      "optional": true
    },
    "@types/react-dom": {
      "optional": true
    }
  }
}
