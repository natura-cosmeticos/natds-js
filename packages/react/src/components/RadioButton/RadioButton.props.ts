import { BrandTypes } from '../../brandTypes/brandTypes'

export interface RadioButtonProps {

  /**
   * Specifies a description to your radio.
   * Use it in cases where a text label is not visible on the screen.
   */
  ariaLabel?: string
  /**
   * If `true`, the RadioButton is checked
   * @default false
   */
  checked?: boolean

  /**
   * Specify an optional className to be added to your Badge
   */
  className?: string
  /**
   * Specify an optional Brand to be added to your Badge
   */
  brand?: BrandTypes

  /**
   * If `true`, the RadioButton will be disabled.
   * @default false
   */
  disabled?: boolean

  /**
   * The id of the RadioButton element.
   * If you are using the 'labelText' attribute, this value it will automatically pass to it
   * Otherwise, remember to provide the same value for your own label
   */
  id: string

  /**
   * Label that appears to the right of the component
   */
  label?: string

  /**
   * Name attribute of the RadioButton element.
   */
  name?: string

  /**
  * The onBlur event occurs the element loses focus.
  */
  onBlur?: React.ChangeEventHandler<HTMLInputElement>

  /**
  * The onChange event occurs when the checked state has been changed.
  */
  onChange: React.ChangeEventHandler<HTMLInputElement>

  /**
  * The onFocus event occurs the element gets focus.
  */
  onFocus?: React.ChangeEventHandler<HTMLInputElement>

  /**
   * If `true`, the RadioButton will be required.
   * @default false
   */
  required?: boolean;

  /**
   * Optional ID for testing
   */
  testID?: string,

  /**
   * The string to use as the value of the RadioButton when submitting the form,
   * if the RadioButton is currently toggled on
   */
  value: string
}
