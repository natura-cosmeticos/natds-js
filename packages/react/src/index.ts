export * from './ThemeProvider'
export * from './growthPlanColors/growthPlanColors'
export { useTheme, createUseStyles } from 'react-jss'

export * from './components/AppBarTop'
export * from './components/Avatar'
export * from './components/Badge'
export * from './components/Button'
export * from './components/Card'
export * from './components/Checkbox'
export * from './components/Collapse'
export * from './components/Counter'
export * from './components/Dialog'
export * from './components/Divider'
export * from './components/Drawer'
export * from './components/Icon'
export * from './components/IconButton'
export * from './components/Input'
export * from './components/Label'
export * from './components/Link'
export * from './components/ListItem'
export * from './components/Logo'
export * from './components/MenuItem'
export * from './components/ProgressIndicator'
export * from './components/RadioButton'
export * from './components/Rating'
export * from './components/Shortcut'
export * from './components/TextField'
export * from './components/Tab'
export * from './components/TabItem'
export * from './components/Alert'
export * from './components/Select'
export * from './components/Switch'
export * from './components/Tag'
export * from './components/Chip'
export * from './components/Typography'
export * from './components/AccordionGroup'
export * from './components/AccordionItem'
export * from './components/Image'
export * from './components/Snackbar'
export * from './components/Autocomplete'
export * from './components/GayaButton'
