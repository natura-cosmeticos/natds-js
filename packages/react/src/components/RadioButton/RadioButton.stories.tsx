import React, { useState } from 'react'
import { Story, Meta } from '@storybook/react'
import { RadioButton, RadioButtonProps } from '.'

const componentStatus = `
> RadioButton allow users to select one option from a set.

## Properties
| Property                      | Values                           |    Status           |
|---                            |                               ---|                  ---|
| **variant(no prop)**          | Standard                         | ✅ Available        |
| **disabled**                  | true/false                       | ✅ Available        |
| **id**                        | string                           | ✅ Available        |
| **label**                     | string                           | ✅ Available        |
| **name**                      | string                           | ✅ Available        |
| **onBlur**                    | function                         | ✅ Available        |
| **onChange**                  | function                         | ✅ Available        |
| **onFocus**                   | function                         | ✅ Available        |
| **required**                  | true/false                       | ✅ Available        |
| **value**                     | string                           | ✅ Available        |
| **brand**                     | avon, avon_v2, natura, natura_v2, theBodyShop, <br /> consultoriaDeBeleza, casaEestilo            | ✅ Available        |

## Technical Usages Examples
`

export default {
  title: 'Components/RadioButton',
  component: RadioButton,
  parameters: {
    componentSubtitle: '',
    docs: { description: { component: componentStatus } },
    actions: { argTypesRegex: '^on.*' }
  }
} as Meta

export const Playground: Story<RadioButtonProps> = ({ checked, ...args }) => {
  const [check, setCheck] = useState(checked)

  return (

    <RadioButton {...args} checked={check} onChange={() => setCheck(!check)} />

  )
}
Playground.args = { checked: false, label: 'Example', ariaLabel: 'Example' }

export const Choose: Story<RadioButtonProps> = ({ checked, ...args }) => {
  const showFocus = checked ? 0 : 1
  const [check, setCheck] = useState(showFocus)

  return (
    <>

      <RadioButton {...args} label="Example1" checked={check === 1} onChange={() => setCheck(1)} />
      <RadioButton {...args} label="Example2" checked={check === 2} onChange={() => setCheck(2)} />
      <RadioButton {...args} label="Example3" checked={check === 3} onChange={() => setCheck(3)} />

    </>
  )
}
Choose.args = { checked: false, label: 'Example' }

export const Disabled: Story<RadioButtonProps> = (args) => (
  <>
    <RadioButton {...args} />
    <RadioButton {...args} checked />
  </>
)
Disabled.args = { ...Playground.args, disabled: true }
