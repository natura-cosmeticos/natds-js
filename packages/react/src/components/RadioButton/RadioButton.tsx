/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable max-lines-per-function */
import React, { useRef } from 'react'
import { Ripple } from '../Ripple'
import { Label } from '../Label'
import { RadioButtonProps } from './RadioButton.props'
import styles from './RadioButton.styles'

const RadioButton = React.forwardRef<HTMLInputElement, RadioButtonProps>(({
  checked = false,
  ariaLabel,
  className = '',
  disabled = false,
  id,
  brand,
  label,
  name,
  onBlur,
  onChange,
  onFocus,
  required = false,
  testID = `ds-RadioButton-${id}`,
  value
}, ref) => {
  const {
    radioButton, wrapper, container, labelText
  } = styles({ disabled, brand })

  const getRippleColor = checked && !disabled ? 'primary' : 'highlight'
  const inputRef = useRef<HTMLInputElement>(null)

  const handleRippleClick = (event: React.MouseEvent<HTMLDivElement>) => {
    if (disabled) {
      return
    }

    if (onChange) {
      if (inputRef.current) {
        inputRef.current.focus()
      }
      onChange(event as any)
    }
  }

  return (
    <div className={`${className} ${container} RadioButton-gaya`}>
      <Ripple
        brand={brand}
        isCentered
        hideOverflow={false}
        disabled={disabled}
        color={getRippleColor}
        animationDuration={500}
        showHover
        radius="50%"
        onClick={handleRippleClick}
        focus
      >
        <div className={wrapper}>
          <input
            aria-label={ariaLabel}
            checked={checked}
            className={radioButton}
            data-testid={testID}
            disabled={disabled}
            id={id}
            name={name}
            onBlur={onBlur}
            onFocus={onFocus}
            onChange={onChange}
            type="radio"
            value={value}
            required={required}
            ref={ref || inputRef}
          />
        </div>
      </Ripple>
      {label && <Label className={labelText} label={label} htmlFor={id} />}
    </div>
  )
})

export default RadioButton
