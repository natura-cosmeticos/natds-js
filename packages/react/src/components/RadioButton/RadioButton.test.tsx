import React from 'react'
import { fireEvent } from '@testing-library/react'
import renderWithTheme from '../../helpers/renderWithTheme'
import RadioButton from './RadioButton'
import { RadioButtonProps } from './RadioButton.props'

const defaultProps: RadioButtonProps = {
  onChange: () => '',
  id: 'any',
  value: 'any value'
}

describe('RadioButton', () => {
  it('should render correctly with default props', () => {
    const { styles, component } = renderWithTheme(<RadioButton {...defaultProps} />)

    expect(component.getByTestId('ds-RadioButton-any')).not.toBeChecked()
    expect(component.getByTestId('ds-RadioButton-any')).not.toBeDisabled()
    expect([styles, component.container]).toMatchSnapshot()
  })

  it('should render correctly when is checked', () => {
    const { styles, component } = renderWithTheme(<RadioButton {...defaultProps} checked />)

    expect(component.getByTestId('ds-RadioButton-any')).toBeChecked()
    expect([styles, component]).toMatchSnapshot()
  })

  it('should render correctly when is disabled', () => {
    const { styles, component } = renderWithTheme(<RadioButton {...defaultProps} disabled />)

    expect(component.getByTestId('ds-RadioButton-any')).toBeDisabled()
    expect([styles, component.container]).toMatchSnapshot()
  })

  it('should call onChange', () => {
    const onChangeMock = jest.fn()
    const { component } = renderWithTheme(<RadioButton {...defaultProps} onChange={onChangeMock} />)

    fireEvent.click(component.getByTestId('ds-RadioButton-any'))
    expect(onChangeMock).toHaveBeenCalledTimes(2)
  })

  it('should render correctly with label', () => {
    const { styles, component } = renderWithTheme(<RadioButton {...defaultProps} label="Example" />)

    expect([styles, component.container]).toMatchSnapshot()
  })
})
