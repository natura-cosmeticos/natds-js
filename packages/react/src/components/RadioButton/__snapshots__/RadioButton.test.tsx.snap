// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton should render correctly when is checked 1`] = `
Array [
  ".sharedRippleEffect-0-2-7 {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
}
.wrapper-0-2-8 {
  height: fit-content;
  outline: none;
  position: relative;
}
.wrapper-0-2-8:focus:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0.16;
  z-index: -1;
  position: absolute;
}
.wrapper-0-2-8:hover:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
}
.ripple-0-2-9 {
  opacity: 0;
  position: absolute;
  transform: translate(-50%, -50%);
  border-radius: 50%;
}
.rippleContainer-0-2-10 {
  inset: 0;
  position: absolute;
}
.rippleActive-0-2-11 {
  z-index: 99;
}
@keyframes keyframes-ripple-0-2-12 {
  0% {
    transform: translate(-50%, -50%) scale(0);
  }
  75% {
    opacity: 0.12;
    transform: translate(-50%, -50%) scale(1);
  }
  100% {
    opacity: 0;
    z-index: 0;
  }
}
  .wrapper-d0-0-2-13 {
    cursor: pointer;
    display: inline-block;
    align-self: start;
  }
  .wrapper-d0-0-2-13:focus:after {
    background-color: #f4ab34;
    border-radius: 50%;
  }
  .wrapper-d0-0-2-13:hover:after {
    background-color: #f4ab34;
    opacity: 0.16;
    border-radius: 50%;
  }
  .ripple-d1-0-2-14 {
    background-color: #f4ab34;
    height: 0;
    left: 50%;
    top: 50%;
    transition: scale opacity 500ms;
    width: 0;
  }
  .rippleContainer-d2-0-2-15 {
    border-radius: 4px;
  }
  .rippleActive-d3-0-2-16 {
    animation: keyframes-ripple-0-2-12 500ms;
  }
  .wrapper-0-2-1 {
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
  }
  .radioButton-0-2-2 {
    width: 18px;
    border: 2px solid #737373;
    cursor: pointer;
    height: 18px;
    margin: 0;
    display: inline-block;
    position: relative;
    appearance: none;
    border-radius: 12px;
  }
  .radioButton-0-2-2:focus {
    outline: none;
  }
  .radioButton-0-2-2:disabled {
    border: 2px solid #8c8c8c;
    cursor: default;
  }
  .radioButton-0-2-2:checked {
    background-color: #ffffff;
  }
  .radioButton-0-2-2:checked:after {
    top: 50%;
    left: 50%;
    width: 8px;
    height: 8px;
    content: \\"\\";
    display: inline-block;
    position: absolute;
    font-size: 18px;
    transform: translate(-50%, -50%);
    text-align: center;
    border-radius: 50%;
  }
  .radioButton-0-2-2:checked:disabled {
    border: 2px solid #8c8c8c;
  }
  .container-0-2-3 {
    display: flex;
    align-items: center;
  }
  .labelText-0-2-4 {
    font-size: 14px;
    font-family: Roboto, Arial;
    font-weight: 400;
    line-height: 1.5;
    letter-spacing: 0.22px;
  }
  .radioButton-d0-0-2-5 {  }
  .radioButton-d0-0-2-5:checked {
    border: 2px solid #f4ab34;
  }
  .radioButton-d0-0-2-5:checked:after {
    background-color: #f4ab34;
  }
  .labelText-d1-0-2-6 {
    color: #363636;
  }",
  Object {
    "asFragment": [Function],
    "baseElement": <body>
      <div>
        <div
          class=" container-0-2-3 RadioButton-gaya"
        >
          <div
            class="wrapper-0-2-8 wrapper-d0-0-2-13"
            data-testid="ripple-wrapper"
            tabindex="0"
          >
            <div
              class="rippleContainer-0-2-10 rippleContainer-d2-0-2-15"
            >
              <div
                class="ripple-0-2-9 ripple-d1-0-2-14 "
                data-testid="ripple-animation"
              />
            </div>
            <div
              class="wrapper-0-2-1"
            >
              <input
                checked=""
                class="radioButton-0-2-2 radioButton-d0-0-2-5"
                data-testid="ds-RadioButton-any"
                id="any"
                type="radio"
                value="any value"
              />
            </div>
          </div>
        </div>
      </div>
    </body>,
    "container": <div>
      <div
        class=" container-0-2-3 RadioButton-gaya"
      >
        <div
          class="wrapper-0-2-8 wrapper-d0-0-2-13"
          data-testid="ripple-wrapper"
          tabindex="0"
        >
          <div
            class="rippleContainer-0-2-10 rippleContainer-d2-0-2-15"
          >
            <div
              class="ripple-0-2-9 ripple-d1-0-2-14 "
              data-testid="ripple-animation"
            />
          </div>
          <div
            class="wrapper-0-2-1"
          >
            <input
              checked=""
              class="radioButton-0-2-2 radioButton-d0-0-2-5"
              data-testid="ds-RadioButton-any"
              id="any"
              type="radio"
              value="any value"
            />
          </div>
        </div>
      </div>
    </div>,
    "debug": [Function],
    "findAllByAltText": [Function],
    "findAllByDisplayValue": [Function],
    "findAllByLabelText": [Function],
    "findAllByPlaceholderText": [Function],
    "findAllByRole": [Function],
    "findAllByTestId": [Function],
    "findAllByText": [Function],
    "findAllByTitle": [Function],
    "findByAltText": [Function],
    "findByDisplayValue": [Function],
    "findByLabelText": [Function],
    "findByPlaceholderText": [Function],
    "findByRole": [Function],
    "findByTestId": [Function],
    "findByText": [Function],
    "findByTitle": [Function],
    "getAllByAltText": [Function],
    "getAllByDisplayValue": [Function],
    "getAllByLabelText": [Function],
    "getAllByPlaceholderText": [Function],
    "getAllByRole": [Function],
    "getAllByTestId": [Function],
    "getAllByText": [Function],
    "getAllByTitle": [Function],
    "getByAltText": [Function],
    "getByDisplayValue": [Function],
    "getByLabelText": [Function],
    "getByPlaceholderText": [Function],
    "getByRole": [Function],
    "getByTestId": [Function],
    "getByText": [Function],
    "getByTitle": [Function],
    "queryAllByAltText": [Function],
    "queryAllByDisplayValue": [Function],
    "queryAllByLabelText": [Function],
    "queryAllByPlaceholderText": [Function],
    "queryAllByRole": [Function],
    "queryAllByTestId": [Function],
    "queryAllByText": [Function],
    "queryAllByTitle": [Function],
    "queryByAltText": [Function],
    "queryByDisplayValue": [Function],
    "queryByLabelText": [Function],
    "queryByPlaceholderText": [Function],
    "queryByRole": [Function],
    "queryByTestId": [Function],
    "queryByText": [Function],
    "queryByTitle": [Function],
    "rerender": [Function],
    "unmount": [Function],
  },
]
`;

exports[`RadioButton should render correctly when is disabled 1`] = `
Array [
  ".sharedRippleEffect-0-2-7 {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
}
.wrapper-0-2-8 {
  height: fit-content;
  outline: none;
  position: relative;
}
.wrapper-0-2-8:focus:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0.16;
  z-index: -1;
  position: absolute;
}
.wrapper-0-2-8:hover:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
}
.ripple-0-2-9 {
  opacity: 0;
  position: absolute;
  transform: translate(-50%, -50%);
  border-radius: 50%;
}
.rippleContainer-0-2-10 {
  inset: 0;
  position: absolute;
}
.rippleActive-0-2-11 {
  z-index: 99;
}
@keyframes keyframes-ripple-0-2-12 {
  0% {
    transform: translate(-50%, -50%) scale(0);
  }
  75% {
    opacity: 0.12;
    transform: translate(-50%, -50%) scale(1);
  }
  100% {
    opacity: 0;
    z-index: 0;
  }
}
  .wrapper-d0-0-2-13 {
    cursor: default;
    display: inline-block;
    align-self: start;
  }
  .wrapper-d0-0-2-13:focus:after {
    border-radius: 50%;
  }
  .wrapper-d0-0-2-13:hover:after {
    background-color: #111111;
    opacity: 0;
    border-radius: 50%;
  }
  .ripple-d1-0-2-14 {
    background-color: #111111;
    height: 0;
    left: 50%;
    top: 50%;
    transition: scale opacity 500ms;
    width: 0;
  }
  .rippleContainer-d2-0-2-15 {
    border-radius: 4px;
  }
  .rippleActive-d3-0-2-16 {
    animation: keyframes-ripple-0-2-12 500ms;
  }
  .wrapper-0-2-1 {
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
  }
  .radioButton-0-2-2 {
    width: 18px;
    border: 2px solid #737373;
    cursor: pointer;
    height: 18px;
    margin: 0;
    display: inline-block;
    position: relative;
    appearance: none;
    border-radius: 12px;
  }
  .radioButton-0-2-2:focus {
    outline: none;
  }
  .radioButton-0-2-2:disabled {
    border: 2px solid #8c8c8c;
    cursor: default;
  }
  .radioButton-0-2-2:checked {
    background-color: #ffffff;
  }
  .radioButton-0-2-2:checked:after {
    top: 50%;
    left: 50%;
    width: 8px;
    height: 8px;
    content: \\"\\";
    display: inline-block;
    position: absolute;
    font-size: 18px;
    transform: translate(-50%, -50%);
    text-align: center;
    border-radius: 50%;
  }
  .radioButton-0-2-2:checked:disabled {
    border: 2px solid #8c8c8c;
  }
  .container-0-2-3 {
    display: flex;
    align-items: center;
  }
  .labelText-0-2-4 {
    font-size: 14px;
    font-family: Roboto, Arial;
    font-weight: 400;
    line-height: 1.5;
    letter-spacing: 0.22px;
  }
  .radioButton-d0-0-2-5 {  }
  .radioButton-d0-0-2-5:checked {
    border: 2px solid #f4ab34;
  }
  .radioButton-d0-0-2-5:checked:after {
    background-color: #8c8c8c;
  }
  .labelText-d1-0-2-6 {
    color: #8c8c8c;
  }",
  <div>
    <div
      class=" container-0-2-3 RadioButton-gaya"
    >
      <div
        class="wrapper-0-2-8 wrapper-d0-0-2-13"
        data-testid="ripple-wrapper"
        tabindex="0"
      >
        <div
          class="rippleContainer-0-2-10 rippleContainer-d2-0-2-15"
        >
          <div
            class="ripple-0-2-9 ripple-d1-0-2-14 "
            data-testid="ripple-animation"
          />
        </div>
        <div
          class="wrapper-0-2-1"
        >
          <input
            class="radioButton-0-2-2 radioButton-d0-0-2-5"
            data-testid="ds-RadioButton-any"
            disabled=""
            id="any"
            type="radio"
            value="any value"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`RadioButton should render correctly with default props 1`] = `
Array [
  ".sharedRippleEffect-0-2-7 {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
}
.wrapper-0-2-8 {
  height: fit-content;
  outline: none;
  position: relative;
}
.wrapper-0-2-8:focus:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0.16;
  z-index: -1;
  position: absolute;
}
.wrapper-0-2-8:hover:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
}
.ripple-0-2-9 {
  opacity: 0;
  position: absolute;
  transform: translate(-50%, -50%);
  border-radius: 50%;
}
.rippleContainer-0-2-10 {
  inset: 0;
  position: absolute;
}
.rippleActive-0-2-11 {
  z-index: 99;
}
@keyframes keyframes-ripple-0-2-12 {
  0% {
    transform: translate(-50%, -50%) scale(0);
  }
  75% {
    opacity: 0.12;
    transform: translate(-50%, -50%) scale(1);
  }
  100% {
    opacity: 0;
    z-index: 0;
  }
}
  .wrapper-d0-0-2-13 {
    cursor: pointer;
    display: inline-block;
    align-self: start;
  }
  .wrapper-d0-0-2-13:focus:after {
    background-color: #111111;
    border-radius: 50%;
  }
  .wrapper-d0-0-2-13:hover:after {
    background-color: #111111;
    opacity: 0.16;
    border-radius: 50%;
  }
  .ripple-d1-0-2-14 {
    background-color: #111111;
    height: 0;
    left: 50%;
    top: 50%;
    transition: scale opacity 500ms;
    width: 0;
  }
  .rippleContainer-d2-0-2-15 {
    border-radius: 4px;
  }
  .rippleActive-d3-0-2-16 {
    animation: keyframes-ripple-0-2-12 500ms;
  }
  .wrapper-0-2-1 {
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
  }
  .radioButton-0-2-2 {
    width: 18px;
    border: 2px solid #737373;
    cursor: pointer;
    height: 18px;
    margin: 0;
    display: inline-block;
    position: relative;
    appearance: none;
    border-radius: 12px;
  }
  .radioButton-0-2-2:focus {
    outline: none;
  }
  .radioButton-0-2-2:disabled {
    border: 2px solid #8c8c8c;
    cursor: default;
  }
  .radioButton-0-2-2:checked {
    background-color: #ffffff;
  }
  .radioButton-0-2-2:checked:after {
    top: 50%;
    left: 50%;
    width: 8px;
    height: 8px;
    content: \\"\\";
    display: inline-block;
    position: absolute;
    font-size: 18px;
    transform: translate(-50%, -50%);
    text-align: center;
    border-radius: 50%;
  }
  .radioButton-0-2-2:checked:disabled {
    border: 2px solid #8c8c8c;
  }
  .container-0-2-3 {
    display: flex;
    align-items: center;
  }
  .labelText-0-2-4 {
    font-size: 14px;
    font-family: Roboto, Arial;
    font-weight: 400;
    line-height: 1.5;
    letter-spacing: 0.22px;
  }
  .radioButton-d0-0-2-5 {  }
  .radioButton-d0-0-2-5:checked {
    border: 2px solid #f4ab34;
  }
  .radioButton-d0-0-2-5:checked:after {
    background-color: #f4ab34;
  }
  .labelText-d1-0-2-6 {
    color: #363636;
  }",
  <div>
    <div
      class=" container-0-2-3 RadioButton-gaya"
    >
      <div
        class="wrapper-0-2-8 wrapper-d0-0-2-13"
        data-testid="ripple-wrapper"
        tabindex="0"
      >
        <div
          class="rippleContainer-0-2-10 rippleContainer-d2-0-2-15"
        >
          <div
            class="ripple-0-2-9 ripple-d1-0-2-14 "
            data-testid="ripple-animation"
          />
        </div>
        <div
          class="wrapper-0-2-1"
        >
          <input
            class="radioButton-0-2-2 radioButton-d0-0-2-5"
            data-testid="ds-RadioButton-any"
            id="any"
            type="radio"
            value="any value"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`RadioButton should render correctly with label 1`] = `
Array [
  ".sharedRippleEffect-0-2-7 {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
}
.wrapper-0-2-8 {
  height: fit-content;
  outline: none;
  position: relative;
}
.wrapper-0-2-8:focus:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0.16;
  z-index: -1;
  position: absolute;
}
.wrapper-0-2-8:hover:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
}
.ripple-0-2-9 {
  opacity: 0;
  position: absolute;
  transform: translate(-50%, -50%);
  border-radius: 50%;
}
.rippleContainer-0-2-10 {
  inset: 0;
  position: absolute;
}
.rippleActive-0-2-11 {
  z-index: 99;
}
@keyframes keyframes-ripple-0-2-12 {
  0% {
    transform: translate(-50%, -50%) scale(0);
  }
  75% {
    opacity: 0.12;
    transform: translate(-50%, -50%) scale(1);
  }
  100% {
    opacity: 0;
    z-index: 0;
  }
}
  .wrapper-d0-0-2-13 {
    cursor: pointer;
    display: inline-block;
    align-self: start;
  }
  .wrapper-d0-0-2-13:focus:after {
    background-color: #111111;
    border-radius: 50%;
  }
  .wrapper-d0-0-2-13:hover:after {
    background-color: #111111;
    opacity: 0.16;
    border-radius: 50%;
  }
  .ripple-d1-0-2-14 {
    background-color: #111111;
    height: 0;
    left: 50%;
    top: 50%;
    transition: scale opacity 500ms;
    width: 0;
  }
  .rippleContainer-d2-0-2-15 {
    border-radius: 4px;
  }
  .rippleActive-d3-0-2-16 {
    animation: keyframes-ripple-0-2-12 500ms;
  }
  .labelContainer-0-2-17 {
    color: #111111;
    font-family: \\"Roboto\\", Arial;
  }
  .wrapper-0-2-1 {
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
  }
  .radioButton-0-2-2 {
    width: 18px;
    border: 2px solid #737373;
    cursor: pointer;
    height: 18px;
    margin: 0;
    display: inline-block;
    position: relative;
    appearance: none;
    border-radius: 12px;
  }
  .radioButton-0-2-2:focus {
    outline: none;
  }
  .radioButton-0-2-2:disabled {
    border: 2px solid #8c8c8c;
    cursor: default;
  }
  .radioButton-0-2-2:checked {
    background-color: #ffffff;
  }
  .radioButton-0-2-2:checked:after {
    top: 50%;
    left: 50%;
    width: 8px;
    height: 8px;
    content: \\"\\";
    display: inline-block;
    position: absolute;
    font-size: 18px;
    transform: translate(-50%, -50%);
    text-align: center;
    border-radius: 50%;
  }
  .radioButton-0-2-2:checked:disabled {
    border: 2px solid #8c8c8c;
  }
  .container-0-2-3 {
    display: flex;
    align-items: center;
  }
  .labelText-0-2-4 {
    font-size: 14px;
    font-family: Roboto, Arial;
    font-weight: 400;
    line-height: 1.5;
    letter-spacing: 0.22px;
  }
  .radioButton-d0-0-2-5 {  }
  .radioButton-d0-0-2-5:checked {
    border: 2px solid #f4ab34;
  }
  .radioButton-d0-0-2-5:checked:after {
    background-color: #f4ab34;
  }
  .labelText-d1-0-2-6 {
    color: #363636;
  }",
  <div>
    <div
      class=" container-0-2-3 RadioButton-gaya"
    >
      <div
        class="wrapper-0-2-8 wrapper-d0-0-2-13"
        data-testid="ripple-wrapper"
        tabindex="0"
      >
        <div
          class="rippleContainer-0-2-10 rippleContainer-d2-0-2-15"
        >
          <div
            class="ripple-0-2-9 ripple-d1-0-2-14 "
            data-testid="ripple-animation"
          />
        </div>
        <div
          class="wrapper-0-2-1"
        >
          <input
            class="radioButton-0-2-2 radioButton-d0-0-2-5"
            data-testid="ds-RadioButton-any"
            id="any"
            type="radio"
            value="any value"
          />
        </div>
      </div>
      <label
        class="labelText-0-2-4 labelText-d1-0-2-6 labelContainer-0-2-17"
        for="any"
      >
        Example
      </label>
    </div>
  </div>,
]
`;
